tritium-cache/src/main/java/com/palantir/tritium/cache/Cache.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.cache;
+
+import com.google.errorprone.annotations.CheckReturnValue;
+import com.google.errorprone.annotations.CompileTimeConstant;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+
+public enum Cache {
+    ;
+
+    @CheckReturnValue
+    public static <K, V> NameBuilder<K, V> builder() {
+        return new CacheBuilder<>();
+    }
+
+    @CheckReturnValue
+    public interface NameBuilder<K, V> {
+
+        SizeBuilder<K, V> name(@CompileTimeConstant String name);
+    }
+
+    @CheckReturnValue
+    public interface SizeBuilder<K, V> {
+
+        ExpiryBuilder<K, V> maximumSize(long maximumSize);
+
+        ExpiryBuilder<K, V> maximumSize(long maximumSize, Weigher<? super K, ? super V> weigher);
+    }
```

Comment:
I avoided providing methods that take a `Refreshable<Long> maximumSize` for a couple of reasons:

- Avoid adding a `refreshable` dependency.

- Simplify the implementation.

    Updating the maximum size requires creating a refreshable subscriber whose lifetime we have to manage. It's not clear how we would do that unless we make the entire `Cache` `AutoCloseable`, which is sort of weird.

- It's not clear to me that there is a real need for this.

    There are only a handful of places internally where we have live-reloaded maximum sizes, and I suspect that we rarely change the maximum size at runtime. Most of those places would probably be fine with fixed maximum cache sizes.

Reply:
Agreed we probably don't want to manage the lifetime & cleanup of Refreshable, and the places currently using live reloading policies can continue just using Caffeine directly.

@Shibi-bala for SA